rir/src/compiler/pir/instruction.h

```diff
@@ -895,10 +896,21 @@ class FLI(Is, 1, Effect::None, EnvAccess::None) {
     void printArgs(std::ostream& out, bool tty) const override;
 };
 
-class FLI(IsObject, 1, Effect::None, EnvAccess::None) {
+class FLI(TypeTest, 1, Effect::None, EnvAccess::None) {
```

Comment:
I don't particularly like that you combined those two instructions.

It feels like those two instructions are somewhat different. `isObject` is testing the R type. `isStub` is testing the internal data representation of a value, that has always type environment.

So in the case of testing for the stub, we are not actually doing a type test. This can also be seen by the fact that  environment and environment stub have the same type in pir. Otoh if the static pir type says `!maybeObject` then we can constantfold isObject.

Reply:
You have a point for sure that I did not think about. Oth, they are very similar instructions at the pir level. I do not see a strong conceptual difference at the pir level and adding instructions for every type of test that we eventually add I think may be nonsense. Pir does not matter what kind of test. Just runs a "predicate" (`isTest`) on a value. Maybe I could change the name from `TypeTest` to `IsTest`.  For the very few cases we need to distinguish (like the constant folding you mention) there is the testFor enum. Then, when translating back to pir I am honoring the difference. 

Summarizing, I admit you have a point. Still, rolling back may imply some time and I am certainly not sure there is a problem here. Neither that this would not be useful in the short term. That said, honestly, if you are strongly against I would gladly rollback.

Reply:
One thing I noticed reviewing this PR is that you have a lot of `isTest->arg<0>().val()` and then even casting it, whereas it would be a lot better to have `isTest->env()`. you decide